corehq/apps/sms/mixin.py

```diff
@@ -138,26 +142,43 @@ def by_suffix(cls, phone_number, include_pending=False):
         Used to lookup a VerifiedNumber, trying to exclude country code digits.
         """
         try:
-            result = cls.phone_lookup(
-                "phone_numbers/verified_number_by_suffix",
-                phone_number,
-                include_pending
-            )
+            result = cls._by_suffix(phone_number)
+            return cls._filter_pending(result, include_pending)
         except MultipleResultsFound:
             # We can't pinpoint who the number belongs to because more than one
             # suffix matches. So treat it as if the result was not found.
-            result = None
-        return result
+            return None
+
+    @classmethod
+    @quickcache(['phone_number'], timeout=60 * 60)
```

Comment:
for now i left these limits kind of low. there's a lot of tests to make sure the cache clearing is working, but i just want to make sure there's no issues before moving this to a higher number like a day or a few days